Framework/DataHandling/src/SaveNexusProcessed.cpp

```diff
@@ -639,18 +623,37 @@ void SaveNexusProcessed::saveSpectraMapNexus(
         detPos[3 * i + 2] = Phi;
       }
     } else
-      for (int i = 0; i < 3 * ndet; i++)
+      for (size_t i = 0; i < 3 * nDetectors; i++)
         detPos[i] = 0.;
 
-    dims[0] = ndet;
+    dims.front() = static_cast<int>(nDetectors);
     dims.push_back(3);
-    dims[1] = 3;
```

Comment:
Why does dims[1] no longer need to be set?

Reply:
It is set by the `push_back(3)` call above.